TechnicalDocuments/Proposals/BuildersToFunctions.md

```diff
@@ -0,0 +1,158 @@
+
+The first point is about simplicity and clean-up of multiple ways of creating screens, while the second point focuses on discoverability and development ergonomics.   
+
+## Proposed solution
+
+**Note:** The proposed solution assumes that a way to authenticate HTTP requests is available at `Current` level (in this case something in the shape of `MainUserSession` abstracted via `UserSession`). 
+
+**Edit** This has voted and approved. An entity will exist at `Current` level.
+
+Let's use `MedicalHistoryBuilder.swift` as an example, since it offers a trivial API: 
+
+``` swift
+init(session: UserSession, visuals: VisualDependenciesProtocol)
+```
+
+And the corresponding public API:
+
+```swift
+func make(modal: Flow) -> UIViewController
+```
+
+Given the Module where it's located it could be migrated to:
+
+```swift
+extension Builder {
+    enum ClinicalRecords {
+        static func makeMedicalHistory(with flow: Flow) -> UIViewController = { _ in UIViewController() }
+    }
+}
+```
+
+Which would allow the consumer to set a default parameter value in the `FlowController` initializer. Instead of:
+
+```swift
+struct ClinicalRecordsFlowController {
+    private let navigation: Flow
+    private let modal: Flow
+    private let presenting: Flow
+    private let builders: ClinicalRecordsChildBuilders
+
+    public init(navigation: Flow,
+                modal: Flow,
+                presenting: Flow,
+                builders: ClinicalRecordsChildBuilders) {
+        self.navigation = navigation
+        self.modal = modal
+        self.presenting = presenting
+        self.builders = builders
+    }
+```
+
+One would:
+
+```swift
+struct ClinicalRecordsFlowController {
+    private let navigation: Flow
+    private let modal: Flow
+    private let presenting: Flow
+
+    public init(navigation: Flow,
+                modal: Flow,
+                presenting: Flow) {
+        self.navigation = navigation
+        self.modal = modal
+        self.presenting = presenting
+    }
+```
+
+Particular medical histories (e.g. for minors) can also be created, but in this case, the minor session would be passed instead of relying on `Current`. This situation (with child builders) can be observed at `ClinicalRecordsChildBuilders.swift`:
+
+```swift
+extension Builder {
+    struct ClinicalRecords {
+        static func makeMedicalHistory(with session: Session, flow: Flow) -> UIViewController 
+        static func makeMedicalHistory(flow: Flow) -> UIViewController
```

Comment:
If I understand correctly, the minor's is the one with the `session` param, right? 
I'd change the method name to avoid confusion defining better the difference.

Reply:
yeah you are correct.